src/main/java/org/terasology/spawning/nui/SpawnScreenLayer.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.spawning.nui;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.prefab.PrefabManager;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.logic.players.LocalPlayer;
+import org.terasology.portals.PortalComponent;
+import org.terasology.registry.In;
+import org.terasology.rendering.nui.CoreScreenLayer;
+import org.terasology.rendering.nui.widgets.UIButton;
+import org.terasology.rendering.nui.widgets.UILabel;
+import org.terasology.spawning.Constants;
+import org.terasology.spawning.OreonSpawnComponent;
+import org.terasology.spawning.OreonSpawnEvent;
+
+import java.util.Map;
+
+
+public class SpawnScreenLayer extends CoreScreenLayer {
+
+    private static final Logger logger = LoggerFactory.getLogger(SpawnScreenLayer.class);
+
+    @In
+    private EntityManager entityManager;
+    @In
+    private LocalPlayer localPlayer;
+    @In
+    private PrefabManager prefabManager;
+
+    private UIButton summonOreonBuilderCommand;
+    private UIButton summonOreonGuardCommand;
+    private UIButton summonOreonKingCommand;
+
+    private UILabel builderResourceRequired;
```

Comment:
`guardResourceRequired` -> `guardResourcesRequired`
You don't need a single resource, you (potentially) need a bunch of resources.